tracing/opentracing/grpc.go

```diff
@@ -0,0 +1,56 @@
+package opentracing
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
+)
+
+// ToGRPCRequest returns a grpc RequestFunc that injects an OpenTracing Span
+// found in `ctx` into the grpc Metadata. If no such Span can be found, the
+// RequestFunc is a noop.
+func ToGRPCRequest(tracer opentracing.Tracer) func(ctx context.Context, md *metadata.MD) context.Context {
+	return func(ctx context.Context, md *metadata.MD) context.Context {
+		if span := opentracing.SpanFromContext(ctx); span != nil {
+			// There's nothing we can do with an error here.
+			_ = tracer.Inject(span, opentracing.TextMap, metadataReaderWriter{md})
```

Comment:
See the line above... since I didn't thread a logger through to report internal errors like this, there's nothing useful to be done with the `error`. There's some tradeoff to the simplicity of the API and how much data we can extract about various error cases.